From 511335e4b0d5f0ad8e45cbdf8e8e6b9632783a22 Mon Sep 17 00:00:00 2001
From: Jianqun xu <jay.xu@rock-chips.com>
Date: Fri, 27 Nov 2015 15:28:33 +0800
Subject: [PATCH] arm64: configs: re-work rockchip default configure

Re-work on rockchip_defconfig follow these steps:
- make ARCH=arm64 rockchip_defconfig
- make savedefconfig
- replace rockchip_defconfig with new defcofnig

Change-Id: Id9f598c415cf0a8526851a414f3ae569ced113f5
Signed-off-by: Jianqun xu <jay.xu@rock-chips.com>
---
 arch/arm64/configs/rockchip_defconfig | 167 +-------------------------
 1 file changed, 2 insertions(+), 165 deletions(-)

diff --git a/arch/arm64/configs/rockchip_defconfig b/arch/arm64/configs/rockchip_defconfig
index a8801427e974..475e9a0cec88 100644
--- a/arch/arm64/configs/rockchip_defconfig
+++ b/arch/arm64/configs/rockchip_defconfig
@@ -4,13 +4,11 @@ CONFIG_SYSVIPC=y
 CONFIG_AUDIT=y
 CONFIG_NO_HZ=y
 CONFIG_HIGH_RES_TIMERS=y
-CONFIG_RCU_FAST_NO_HZ=y
 CONFIG_LOG_BUF_SHIFT=19
 CONFIG_CGROUPS=y
 CONFIG_CGROUP_DEBUG=y
 CONFIG_CGROUP_FREEZER=y
 CONFIG_CGROUP_CPUACCT=y
-CONFIG_RESOURCE_COUNTERS=y
 CONFIG_CGROUP_SCHED=y
 CONFIG_RT_GROUP_SCHED=y
 CONFIG_NAMESPACES=y
@@ -37,17 +35,9 @@ CONFIG_ARCH_ROCKCHIP=y
 # CONFIG_ARM64_ERRATUM_832075 is not set
 # CONFIG_CAVIUM_ERRATUM_22375 is not set
 # CONFIG_CAVIUM_ERRATUM_23154 is not set
-CONFIG_BLOCK_RKNAND=y
 CONFIG_SCHED_MC=y
 CONFIG_NR_CPUS=8
-CONFIG_DISABLE_CPU_SCHED_DOMAIN_BALANCE=y
-CONFIG_SCHED_HMP=y
-CONFIG_HMP_FAST_CPU_MASK="4-7"
-CONFIG_HMP_SLOW_CPU_MASK="0-3"
-CONFIG_HMP_VARIABLE_SCALE=y
-CONFIG_HMP_FREQUENCY_INVARIANT_SCALE=y
 CONFIG_PREEMPT=y
-CONFIG_ARMV7_COMPAT=y
 CONFIG_KSM=y
 CONFIG_CMA=y
 CONFIG_ZSMALLOC=y
@@ -59,19 +49,7 @@ CONFIG_PM_AUTOSLEEP=y
 CONFIG_PM_WAKELOCKS=y
 CONFIG_PM_DEBUG=y
 CONFIG_WQ_POWER_EFFICIENT_DEFAULT=y
-CONFIG_SUSPEND_TIME=y
-CONFIG_CPU_FREQ_DEFAULT_GOV_INTERACTIVE=y
-CONFIG_CPU_FREQ_GOV_PERFORMANCE=y
-CONFIG_CPU_FREQ_GOV_POWERSAVE=y
-CONFIG_CPU_FREQ_GOV_USERSPACE=y
-CONFIG_CPU_FREQ_GOV_ONDEMAND=y
-CONFIG_CPU_FREQ_GOV_CONSERVATIVE=y
-CONFIG_ARM_BIG_LITTLE_CPUFREQ=y
-CONFIG_ARM_DT_BL_CPUFREQ=y
-# CONFIG_ARM_ROCKCHIP_CPUFREQ is not set
-CONFIG_ARM_ROCKCHIP_BL_CPUFREQ=y
 CONFIG_CPU_IDLE=y
-CONFIG_ARM64_CPUIDLE=y
 CONFIG_NET=y
 CONFIG_PACKET=y
 CONFIG_UNIX=y
@@ -88,7 +66,6 @@ CONFIG_INET_AH=y
 CONFIG_INET_ESP=y
 CONFIG_INET_IPCOMP=y
 # CONFIG_INET_LRO is not set
-CONFIG_IPV6_PRIVACY=y
 CONFIG_IPV6_ROUTER_PREF=y
 CONFIG_IPV6_ROUTE_INFO=y
 CONFIG_IPV6_OPTIMISTIC_DAD=y
@@ -114,7 +91,6 @@ CONFIG_NF_CONNTRACK_PPTP=y
 CONFIG_NF_CONNTRACK_SANE=y
 CONFIG_NF_CONNTRACK_TFTP=y
 CONFIG_NF_CT_NETLINK=y
-CONFIG_NETFILTER_TPROXY=y
 CONFIG_NETFILTER_XT_TARGET_CLASSIFY=y
 CONFIG_NETFILTER_XT_TARGET_CONNMARK=y
 CONFIG_NETFILTER_XT_TARGET_CONNSECMARK=y
@@ -141,10 +117,7 @@ CONFIG_NETFILTER_XT_MATCH_MARK=y
 CONFIG_NETFILTER_XT_MATCH_MULTIPORT=y
 CONFIG_NETFILTER_XT_MATCH_POLICY=y
 CONFIG_NETFILTER_XT_MATCH_PKTTYPE=y
-CONFIG_NETFILTER_XT_MATCH_QTAGUID=y
 CONFIG_NETFILTER_XT_MATCH_QUOTA=y
-CONFIG_NETFILTER_XT_MATCH_QUOTA2=y
-CONFIG_NETFILTER_XT_MATCH_QUOTA2_LOG=y
 CONFIG_NETFILTER_XT_MATCH_SOCKET=y
 CONFIG_NETFILTER_XT_MATCH_STATE=y
 CONFIG_NETFILTER_XT_MATCH_STATISTIC=y
@@ -152,17 +125,13 @@ CONFIG_NETFILTER_XT_MATCH_STRING=y
 CONFIG_NETFILTER_XT_MATCH_TIME=y
 CONFIG_NETFILTER_XT_MATCH_U32=y
 CONFIG_NF_CONNTRACK_IPV4=y
+CONFIG_NF_NAT_IPV4=y
 CONFIG_IP_NF_IPTABLES=y
 CONFIG_IP_NF_MATCH_AH=y
 CONFIG_IP_NF_MATCH_ECN=y
 CONFIG_IP_NF_MATCH_TTL=y
 CONFIG_IP_NF_FILTER=y
 CONFIG_IP_NF_TARGET_REJECT=y
-CONFIG_IP_NF_TARGET_REJECT_SKERR=y
-CONFIG_NF_NAT_IPV4=y
-CONFIG_IP_NF_TARGET_MASQUERADE=y
-CONFIG_IP_NF_TARGET_NETMAP=y
-CONFIG_IP_NF_TARGET_REDIRECT=y
 CONFIG_IP_NF_MANGLE=y
 CONFIG_IP_NF_RAW=y
 CONFIG_IP_NF_SECURITY=y
@@ -173,7 +142,6 @@ CONFIG_NF_CONNTRACK_IPV6=y
 CONFIG_IP6_NF_IPTABLES=y
 CONFIG_IP6_NF_FILTER=y
 CONFIG_IP6_NF_TARGET_REJECT=y
-CONFIG_IP6_NF_TARGET_REJECT_SKERR=y
 CONFIG_IP6_NF_MANGLE=y
 CONFIG_IP6_NF_RAW=y
 CONFIG_BRIDGE_NF_EBTABLES=y
@@ -204,28 +172,17 @@ CONFIG_BT_HIDP=y
 CONFIG_BT_HCIUART=y
 CONFIG_BT_HCIUART_H4=y
 CONFIG_BT_HCIUART_LL=y
-CONFIG_CFG80211_WEXT=y
 CONFIG_RFKILL=y
-CONFIG_RFKILL_RK=y
 CONFIG_DEVTMPFS=y
 CONFIG_DMA_CMA=y
 CONFIG_ZRAM=y
 CONFIG_ZRAM_LZ4_COMPRESS=y
 CONFIG_BLK_DEV_LOOP=y
-CONFIG_UID_STAT=y
 CONFIG_SRAM=y
-CONFIG_UID_CPUTIME=y
-CONFIG_MPU_SENSORS_TIMERIRQ=y
-CONFIG_INV_SENSORS=y
-CONFIG_MPU_SENSORS_MPU6050B1=y
-CONFIG_MPU_SENSORS_MPU6050_ACCEL=y
-CONFIG_MPU_SENSORS_AK8963=y
 CONFIG_SCSI=y
-CONFIG_SCSI_TGT=y
 CONFIG_BLK_DEV_SD=y
 CONFIG_CHR_DEV_SG=y
 CONFIG_CHR_DEV_SCH=y
-CONFIG_SCSI_MULTI_LUN=y
 CONFIG_SCSI_CONSTANTS=y
 CONFIG_SCSI_LOGGING=y
 CONFIG_SCSI_SCAN_ASYNC=y
@@ -247,7 +204,6 @@ CONFIG_TUN=y
 # CONFIG_NET_VENDOR_SMSC is not set
 # CONFIG_NET_VENDOR_STMICRO is not set
 # CONFIG_NET_VENDOR_WIZNET is not set
-CONFIG_RK_GMAC_ETH=y
 CONFIG_PPP=y
 CONFIG_PPP_BSDCOMP=y
 CONFIG_PPP_DEFLATE=y
@@ -256,8 +212,6 @@ CONFIG_PPP_MPPE=y
 CONFIG_PPP_MULTILINK=y
 CONFIG_PPPOE=y
 CONFIG_PPPOL2TP=y
-CONFIG_PPPOLAC=y
-CONFIG_PPPOPNS=y
 CONFIG_PPP_ASYNC=y
 CONFIG_PPP_SYNC_TTY=y
 CONFIG_SLIP=y
@@ -272,7 +226,6 @@ CONFIG_USB_USBNET=y
 CONFIG_USB_NET_CDC_EEM=y
 CONFIG_USB_NET_CDC_MBIM=y
 CONFIG_USB_NET_DM9601=y
-CONFIG_USB_NET_DM9620=y
 CONFIG_USB_NET_SMSC75XX=y
 CONFIG_USB_NET_SMSC95XX=y
 CONFIG_USB_NET_GL620A=y
@@ -290,13 +243,8 @@ CONFIG_USB_HSO=y
 CONFIG_USB_NET_INT51X1=y
 CONFIG_USB_IPHETH=y
 CONFIG_USB_SIERRA_NET=y
-CONFIG_RTL8188EU=y
-CONFIG_ESP8089=y
-CONFIG_RKWIFI=y
-CONFIG_AP6335=y
 # CONFIG_INPUT_MOUSEDEV is not set
 CONFIG_INPUT_EVDEV=y
-CONFIG_INPUT_KEYRESET=y
 # CONFIG_KEYBOARD_ATKBD is not set
 # CONFIG_INPUT_MOUSE is not set
 CONFIG_INPUT_JOYSTICK=y
@@ -309,25 +257,9 @@ CONFIG_TABLET_USB_AIPTEK=y
 CONFIG_TABLET_USB_GTCO=y
 CONFIG_TABLET_USB_HANWANG=y
 CONFIG_TABLET_USB_KBTAB=y
-CONFIG_TABLET_USB_WACOM=y
 CONFIG_INPUT_TOUCHSCREEN=y
-CONFIG_TOUCHSCREEN_ZET62XX=y
-CONFIG_TOUCHSCREEN_GT9XX=y
-CONFIG_TOUCHSCREEN_VTL_CT36X=y
 CONFIG_INPUT_MISC=y
-CONFIG_INPUT_KEYCHORD=y
-CONFIG_INPUT_RICOH619_PWRKEY=y
 CONFIG_INPUT_UINPUT=y
-CONFIG_INPUT_GPIO=y
-# CONFIG_GS_LIS3DH is not set
-# CONFIG_GS_MMA7660 is not set
-CONFIG_GS_MC3230=y
-# CONFIG_GS_LSM303D is not set
-# CONFIG_COMPASS_DEVICE is not set
-# CONFIG_GYROSCOPE_DEVICE is not set
-# CONFIG_HALL_DEVICE is not set
-CONFIG_ROCKCHIP_REMOTECTL=y
-CONFIG_ROCKCHIP_REMOTECTL_PWM=y
 # CONFIG_SERIO is not set
 # CONFIG_VT is not set
 # CONFIG_LEGACY_PTYS is not set
@@ -335,58 +267,21 @@ CONFIG_SERIAL_8250=y
 # CONFIG_SERIAL_8250_DEPRECATED_OPTIONS is not set
 CONFIG_SERIAL_8250_CONSOLE=y
 CONFIG_SERIAL_8250_DW=y
-CONFIG_SERIAL_ROCKCHIP=y
-CONFIG_SERIAL_ROCKCHIP_CONSOLE=y
 # CONFIG_HW_RANDOM is not set
-CONFIG_I2C=y
 CONFIG_I2C_CHARDEV=y
-CONFIG_I2C_ROCKCHIP=y
-CONFIG_I2C_ROCKCHIP_COMPAT=y
 CONFIG_SPI=y
-CONFIG_SPI_ROCKCHIP_CORE=y
 CONFIG_SPI_ROCKCHIP=y
-CONFIG_SPI_ROCKCHIP_DMA=y
-CONFIG_SPI_ROCKCHIP_TEST=y
 CONFIG_DEBUG_GPIO=y
 CONFIG_GPIO_SYSFS=y
-CONFIG_CHARGER_RT5025=y
-CONFIG_BATTERY_RT5025=y
-CONFIG_CHARGER_RT5036=y
-CONFIG_RT_POWER=y
-CONFIG_BATTERY_RICOH619=y
 CONFIG_TEST_POWER=y
-CONFIG_BATTERY_BQ24296=y
-CONFIG_BATTERY_BQ27320=y
-CONFIG_BATTERY_RK30_ADC_FAC=y
-CONFIG_CW2015_BATTERY=y
-CONFIG_BATTERY_RK818=y
 CONFIG_POWER_RESET_GPIO=y
-CONFIG_POWER_RESET_ROCKCHIP=y
 CONFIG_POWER_AVS=y
 CONFIG_ROCKCHIP_IODOMAIN=y
 CONFIG_THERMAL=y
 CONFIG_ROCKCHIP_THERMAL=y
-CONFIG_MFD_RT5025=y
-CONFIG_MISC_RT5025=y
-CONFIG_IRQ_RT5025=y
-CONFIG_DEBUG_RT5025=y
-CONFIG_MFD_RT5036=y
-CONFIG_MISC_RT5036=y
-CONFIG_IRQ_RT5036=y
 CONFIG_MFD_RK808=y
-CONFIG_MFD_RK818=y
-CONFIG_MFD_RICOH619=y
 CONFIG_REGULATOR=y
 CONFIG_REGULATOR_FIXED_VOLTAGE=y
-CONFIG_REGULATOR_ACT8846=y
-CONFIG_ACT8846_SUPPORT_RESET=y
-CONFIG_REGULATOR_ACT8931=y
-CONFIG_REGULATOR_RT5025=y
-CONFIG_REGULATOR_RT5036=y
-CONFIG_ROCKCHIP_PWM_REGULATOR=y
-CONFIG_REGULATOR_SYR82X=y
-CONFIG_REGULATOR_XZ3216=y
-CONFIG_REGULATOR_RICOH619=y
 CONFIG_MEDIA_SUPPORT=y
 CONFIG_MEDIA_CAMERA_SUPPORT=y
 CONFIG_MEDIA_RC_SUPPORT=y
@@ -395,29 +290,11 @@ CONFIG_VIDEO_V4L2_SUBDEV_API=y
 CONFIG_MEDIA_USB_SUPPORT=y
 CONFIG_USB_VIDEO_CLASS=y
 CONFIG_V4L_PLATFORM_DRIVERS=y
-CONFIG_RK30_CAMERA_ONEFRAME=y
-CONFIG_POWERVR_ROGUE=y
 CONFIG_FB=y
 CONFIG_BACKLIGHT_LCD_SUPPORT=y
 # CONFIG_LCD_CLASS_DEVICE is not set
-CONFIG_BACKLIGHT_CLASS_DEVICE=y
 # CONFIG_BACKLIGHT_GENERIC is not set
 CONFIG_BACKLIGHT_PWM=y
-CONFIG_FB_ROCKCHIP=y
-CONFIG_LCDC_RK3368=y
-CONFIG_LCD_MIPI=y
-CONFIG_RK_TRSM=y
-CONFIG_RK31XX_LVDS=y
-CONFIG_DP_ANX6345=y
-CONFIG_RK32_DP=y
-CONFIG_MIPI_DSI=y
-CONFIG_RK32_MIPI_DSI=y
-CONFIG_RK_HDMI=y
-CONFIG_RK_TVENCODER=y
-CONFIG_RK1000_TVOUT=y
-CONFIG_RK1000_TVOUT_CVBS=y
-CONFIG_ROCKCHIP_RGA=y
-CONFIG_ROCKCHIP_RGA2=y
 CONFIG_LOGO=y
 # CONFIG_LOGO_LINUX_MONO is not set
 # CONFIG_LOGO_LINUX_VGA16 is not set
@@ -430,14 +307,6 @@ CONFIG_SND_VERBOSE_PRINTK=y
 # CONFIG_SND_SPI is not set
 CONFIG_SND_USB_AUDIO=y
 CONFIG_SND_SOC=y
-CONFIG_SND_RK_SOC=y
-CONFIG_SND_RK_SOC_HDMI_I2S=y
-CONFIG_SND_RK_SOC_SPDIF_CARD=y
-CONFIG_SND_RK_SOC_ES8316=y
-CONFIG_SND_RK_SOC_RT5631=y
-CONFIG_SND_RK_SOC_RT3224=y
-CONFIG_SND_RK_SOC_RK1000=y
-CONFIG_SND_RK_SOC_RK312X=y
 CONFIG_HIDRAW=y
 CONFIG_UHID=y
 CONFIG_HID_A4TECH=y
@@ -505,7 +374,6 @@ CONFIG_HID_ZYDACRON=y
 CONFIG_USB_HIDDEV=y
 CONFIG_USB_ANNOUNCE_NEW_DEVICES=y
 CONFIG_USB_EHCI_HCD=y
-CONFIG_USB_EHCI_RK=y
 CONFIG_USB_EHCI_HCD_PLATFORM=y
 CONFIG_USB_OHCI_HCD=y
 CONFIG_USB_OHCI_HCD_PLATFORM=y
@@ -528,67 +396,37 @@ CONFIG_USB_SERIAL=y
 CONFIG_USB_SERIAL_GENERIC=y
 CONFIG_USB_SERIAL_OPTION=y
 CONFIG_USB_TRANCEVIBRATOR=y
-CONFIG_USB_OTG_WAKELOCK=y
 CONFIG_USB_GADGET=y
 CONFIG_USB_GADGET_DEBUG_FILES=y
 CONFIG_USB_GADGET_VBUS_DRAW=500
-CONFIG_USB_G_ANDROID=y
-CONFIG_USB20_HOST=y
-CONFIG_USB20_OTG=y
+CONFIG_USB_ETH=y
 CONFIG_MMC=y
-CONFIG_MMC_PARANOID_SD_INIT=y
 CONFIG_MMC_BLOCK_MINORS=32
 # CONFIG_MMC_BLOCK_BOUNCE is not set
 CONFIG_MMC_DW=y
-CONFIG_MMC_DW_IDMAC=y
 CONFIG_MMC_DW_ROCKCHIP=y
-CONFIG_MMC_DW_SKIP_CACHE_OP=y
 CONFIG_LEDS_GPIO=y
 CONFIG_LEDS_TRIGGERS=y
 CONFIG_LEDS_TRIGGER_BACKLIGHT=y
 CONFIG_LEDS_TRIGGER_DEFAULT_ON=y
-CONFIG_SWITCH=y
 CONFIG_RTC_CLASS=y
-CONFIG_RTC_HYM8563=y
-CONFIG_RTC_RT5036=y
-CONFIG_RTC_DRV_RC5T619=y
 CONFIG_DMADEVICES=y
 CONFIG_PL330_DMA=y
 CONFIG_STAGING=y
 CONFIG_ASHMEM=y
-CONFIG_ANDROID_LOGGER=y
 CONFIG_ANDROID_TIMED_GPIO=y
 CONFIG_ANDROID_LOW_MEMORY_KILLER=y
-CONFIG_ANDROID_INTF_ALARM_DEV=y
 CONFIG_SYNC=y
 CONFIG_SW_SYNC=y
 CONFIG_SW_SYNC_USER=y
 CONFIG_ION=y
-CONFIG_ION_ROCKCHIP=y
-CONFIG_ION_CMA_HIGHMEM=y
-CONFIG_ION_ROCKCHIP_SNAPSHOT=y
-CONFIG_ION_SNAPSHOT_BUF_SHIFT=15
-CONFIG_FIQ_DEBUGGER=y
-CONFIG_FIQ_DEBUGGER_NO_SLEEP=y
-CONFIG_FIQ_DEBUGGER_CONSOLE=y
-CONFIG_FIQ_DEBUGGER_CONSOLE_DEFAULT_ENABLE=y
-CONFIG_FIQ_DEBUGGER_EL3_TO_EL1=y
-CONFIG_COMMON_CLK_DEBUG=y
 # CONFIG_COMMON_CLK_XGENE is not set
 CONFIG_MAILBOX=y
-CONFIG_ROCKCHIP_MAILBOX=y
-CONFIG_SCPI_PROTOCOL=y
-CONFIG_ROCKCHIP_IOMMU=y
-CONFIG_ROCKCHIP_IOVMM=y
 CONFIG_PM_DEVFREQ=y
-CONFIG_ROCKCHIP_RK3368_DDR_FREQ=y
 CONFIG_IIO=y
-CONFIG_ROCKCHIP_ADC=y
 CONFIG_PWM=y
 CONFIG_PWM_ROCKCHIP=y
-CONFIG_RESET_CONTROLLER=y
 CONFIG_PHY_ROCKCHIP_USB=y
-CONFIG_RK_HEADSET=y
 CONFIG_ANDROID=y
 CONFIG_ANDROID_BINDER_IPC=y
 CONFIG_EXT2_FS=y
@@ -614,7 +452,6 @@ CONFIG_NLS_ASCII=y
 CONFIG_NLS_ISO8859_1=y
 CONFIG_NLS_UTF8=y
 CONFIG_PRINTK_TIME=y
-CONFIG_PRINTK_PROCESS=y
 CONFIG_DEBUG_INFO=y
 CONFIG_DEBUG_FS=y
 CONFIG_MAGIC_SYSRQ=y
-- 
2.35.3

